pkg/utils/kubernetes/object.go

```diff
@@ -109,3 +115,37 @@ func CreateResetObjectFunc(obj runtime.Object, scheme *runtime.Scheme) (func(),
 func deepCopyIntoObject(dest, src runtime.Object) {
 	reflect.ValueOf(dest).Elem().Set(reflect.ValueOf(src.DeepCopyObject()).Elem())
 }
+
+// MakeImmutable takes either a *corev1.ConfigMap or a *corev1.Secret object and makes it immutable, i.e., it sets
+// .immutable=true, computes a checksum based on .data, and appends the first 8 characters of the computed checksum
+// to the name of the object. Additionally, it injects the `resources.gardener.cloud/garbage-collectable-reference=true`
+// label.
+func MakeImmutable(obj runtime.Object) error {
+	var (
+		numberOfChecksumChars = 8
+		prependHyphen         = func(name string) string {
+			if strings.HasSuffix(name, "-") {
+				return ""
+			}
+			return "-"
+		}
+		labels = map[string]string{references.LabelKeyGarbageCollectable: references.LabelValueGarbageCollectable}
+	)
+
+	switch o := obj.(type) {
+	case *corev1.Secret:
+		o.Immutable = pointer.Bool(true)
+		o.Name += prependHyphen(o.Name) + utils.ComputeSecretChecksum(o.Data)[:numberOfChecksumChars]
+		o.Labels = utils.MergeStringMaps(o.Labels, labels)
```

Comment:
Why not use `metav1.SetMetaDataLabel(&o.ObjectMeta, references.LabelKeyGarbageCollectable, references.LabelValueGarbageCollectable)`? No need to merge here, right?

Reply:
😄 Right, thanks, done.